### DISCUSSION TITLE
OWL Community Builds

### DISCUSSION THREAD

**Discussion by geezacoleman** [geezacoleman/OpenWeedLocator#75] [General]
If you're building an OWL or have built one already it would be great to learn more about how you plan to/are currently using them and which country you're building it in too!

You could share some photos of the build process, completed versions mounted on equipment or share your experience with the assembly.

We've built them in NSW and WA in Australia for the backs of vehicles and on robots, but it would be neat to learn more about how others might be using the OWL.

**Comment by Minorplanets76** [comment]
Well that's a coincidence! ![image](https://user-images.githubusercontent.com/107326485/233044465-5da8173f-28d8-444d-81f8-0dd4ba56fa0c.jpeg) 
A miracle arrived today from Little Bird Electronics with a new global shutter camera. This will be the start of my Owl test rig. 

I have been running AgOpenGPS for some years along with auto steer and section control on a modest 6m Hardi boom. I recently got a new tractor which is being treated to hydraulic steer valve upgrade and new boards. Once that's implemented I'll be looking at this project. 

I had a look at a Weedseeker unit at a clearing sale recently which reminded me of this project. That was using teejet solenoids on standard teejet bodies. Very expensive yet very simple concept, so I've ordered a selection of solenoids with a view to adapting a similar system to make it easy to give any boom as many sections as it has nozzles. 

I'm in the Gunnedah area NSW. You're welcome to have a look if you don't mind a short detour from the Uni farm sometime.

**Reply by geezacoleman** [reply]
This is great news! Glad to see some Raspberry Pi boards starting to become more available again. Will be great to see how the 6m rig comes together and would also be neat to see it integrated with a AgOpenGPS tractor once you have it up and running.

Do you think you'll use the new libcamera/Picamera2 instead of the old version? That's something we need to get sorted out, the conversion doesn't look too tricky and will future proof the OWL substantially for other camera systems too.

**Reply by Minorplanets76** [reply]
Yes, I will need the new version for the GS cameras. For testing I didn't manage to get it to work using the OWL img  despite quite a bit of updating and installing what Google reckoned I needed.   With a fresh install of the Bullseye release it just works.

Solenoids arrived today. Made me realise how power hungry they can be for a pretty low force.  I'm realising why the commercial ones able to do pwm are expensive.   Will do some tests and see if I can get something working.

**Comment by ahogan27** [comment]
![image](https://github.com/geezacoleman/OpenWeedLocator/assets/23489736/c40bef24-9d50-4c30-97fe-811754d20d0a)

Have finally started on my build. Based in Orange NSW, without immediate access to spraying locations, so am building a bench testing unit. Plan is to just use lights to replicate solenoids. Hoping to contribute to some of the coding, in particular inclusion of GNSS to the system.

**Reply by geezacoleman** [reply]
Very nice set up! Those 12V lights are really handy for testing/diagnosing issues, particularly when you don't want to get water everywhere.

How was the build?

**Reply by ahogan27** [reply]
Build went well. Biggest challenge was getting the camera focused. Didn't realise that the flathead screw on the lens mount was to tighten the back focus adjustment ring, so made the assumption that the back focus had to be screwed all the way in (otherwise it was noticeably loose). Once I figured that out I was able to get the focus sorted quickly. 

Also, didn't seem to be clear in the documentation - what's the command to close out of owl_focus?

**Reply by geezacoleman** [reply]
Good point - I might add that to the documentation. I remember doing that as well. The lens also has somewhat confusing aperture and focus rings that don't seem to match the hold screws.

Should just be able to press ESC after clicking on any of the video windows or Ctrl-C in the terminal. ESC is the neater option though.

**Reply by ahogan27** [reply]
Thanks, for out by Ctrl-C but it didn't feel right, will use ESC next time.

**Comment by Ben-Pankhurst** [comment]
Built an OWL system to attach to a repurposed quad-bike sprayer. Fed from a constant pressure pump with a recirculating feed loop to manage system pressure. We mounted the solenoids onto 4040 aluminium extrusion so we could play around with different nozzle spacing and different spray fan sizes, in combination with different camera combinations. Really have to commend Guy and everyone who has supported OWL for building an amazing system with excellent documentation.
![IMG_4013](https://github.com/geezacoleman/OpenWeedLocator/assets/139436349/b6d2105a-1db6-48a2-8067-1a59a9e6f046)

**Reply by geezacoleman** [reply]
Thanks for sharing Ben! I'm glad the OWL helped kick off the WeedSelector and get you guys started. Keep us updated with how you go!

Any suggestions/improvements for the build/design etc?

**Reply by Ben-Pankhurst** [reply]
Absolutely, will do.

We found the mechanical assembly very straightforward, however it took some poking around and external research to find out RaspiOS Bullseye (and now the newly released RaspiOS Bookworm) are less compatible with the Raspberry Pi HQ Camera. Support for the 'picamera' library has been removed on Bullseye and later, so using Buster fixed our camera detection issues. 

If I was to change anything I would add a section in the software section of the readme explaining what version of RaspiOS to use and why, and potentially a reputable download link as a version of Buster was kind of hard to find online. 

Apart from that, the quality and care that has been put into this project is amazing, keep up the great work!

**Reply by gnaegit** [reply]
I also started with the newly released Raspi OS Bookworm and then had some problems when installing all the packages according to the guide. I managed to install (almost) all of them (or newer versions). But finally when it came to focusing the camera I got an error. I was not sure if it was due to a problem with one of the packages or due to detection of the camera. I could not find the camera in the Raspi configuration under interfaces. Then I changed to Buster as the guide proposes so. However, I still have some problems installing the opencv-contrib-python package as the numpy wheel can not be build. I agree that it would be nice to make it clear what Raspi OS should be used and if newer OS's would be supported in the future.

**Reply by geezacoleman** [reply]
Yes - this is a great point. Thanks for the feedback. The latest RPi OS won't work primarily because of the clash with picamera and picamera2. I would recommend using the existing image and updating it.

Adding picamera2 support and therefore updating it for Bookworm and the Pi 5 is underway. Picamera2 really makes it much easier to use a range of cameras including the Arducam. It's something I hope to have done in February next year - the next couple of months are quite busy unfortunately. In any case, I'll add a clearer note to the installation instructions about this.

**Comment by bassyboi** [comment]
![main boom componet v1 2](https://github.com/user-attachments/assets/b3de8e6c-5b9e-4b39-86ad-a3e2939f8a7c)
![main boom componet v1](https://github.com/user-attachments/assets/3ecb4cf7-b7ca-4818-8998-99a2b4fd9b38)


working on a boom that that will be able to extend size just bolt on and go will come up with a foldable section  this is a 1m part atm

**Reply by bassyboi** [reply]
![boom set-up](https://github.com/user-attachments/assets/159d1400-9038-4c0e-9836-ed3367caf116)

here is them put together

**Reply by bassyboi** [reply]
![utv set up v1](https://github.com/user-attachments/assets/e893e0bf-a80b-455a-ba33-5e859dff4d17)

**Reply by MisterHW** [reply]
Inspired me to have a look around for scaffolding truss modules. There's also a surplus market for them:
![image](https://github.com/user-attachments/assets/8bd66fc4-fd6a-49db-9bd3-0c88b2b5274a)
I presume the average OWL user has means to fabricate their own solution. Sharing this in hopes that it may help lower the barrier of entry regardless. Can see how these would allow to focus on the joint parts and preserve a degree of modularity for future extension / adaption.

**Reply by bassyboi** [reply]
Should work mate!! Yes I was thinking about how I could make it like a boom sprayer that could be made for everyone!! Look if you really want me to I could come up with something if I get the dimensions like pipe wall size and that!!! What size boom you would like!!! I did do a trade as a fabricator/welder so I’ll try make it as easy as I can to fabricate… are the scaffolding pretty common sizes?? 

True about trying to make the entry level as low as possible!! You have definitely given me some ideas

**Reply by bassyboi** [reply]
![12m bar](https://github.com/user-attachments/assets/e1fbc603-3957-4f8f-8d8c-220ebd080214)

bitta promo for the page

**Reply by geezacoleman** [reply]
Epic!! Is that a boom you're going to have built? Truss segments are a great idea @MisterHW for DIY options. For a 12m folding boom I'm trying to figure out how to neatly tuck the units away.

Some low cost, small booms are available like the Croplands MBX series

![20240916_132352](https://github.com/user-attachments/assets/2f637fd9-7d8c-4a77-92e9-d8d26e3a453c)

**Comment by MerbeinRobotics** [comment]
Hi All,

![MR01 011224](https://github.com/user-attachments/assets/b6f17541-296d-418f-8a8e-c4d5cf232d93)

Just thought I'd share a photo of an autonomous robot I've been working on.  I started this project many years ago and this is the latest version of the robot.    It is using Ardurover software and Cube Orange for the autopilot with a Here+ RTK gps unit.  I was always intending on using it for weed spraying in our vineyard (in Merbein, near Mildura, NW Victoria) and when the OWL project came along and though it would complement the robot perfectly.  I built the OWL a couple of years ago and intend on upgrading to the latest hardware and software as well as adding 2 more units so I can cover the full row with of about 3.3 metres.  Testing the robot and the OWL yesterday showed some real promise.

Anyway here is a link to a couple of quick video's of the robot doing some test runs in autonomous mode.

https://photos.app.goo.gl/tXRv7ovMz1C2P35c8

https://photos.app.goo.gl/CJUYMXQkEsf7Ntgw6

Kind Regards,

Stephen

**Comment by Farmpinguin** [comment]
Hi there,
long time reader, first time poster here. 
I´m a farmer from Germany, mostly wheat/barley, canola, sugar beets and corn. I played around with the owl in 2023 but due to the lack of free time i never finished.
Now i´m back and have my first OWL running on the sprayer collecting data for a set of weeds in sugar beets ( mostly bindweed, knotweed and regrown canola)
Between 2023 and now i focused on building spray maps with drones. I worked out that normal DJI drones with RGB-pictures crossed with a VARI-Index work great for finding spots with higher mass of green in row crops. Pair that with a custom QGis model to track those and mark them gave me very detailed spotspray maps.
But now i´m aiming for an autonomous solution (just for test of course, German law is like German people, there´s no fun here), so i´m back to Community. 
<img width="800" height="1421" alt="20260516_101155" src="https://github.com/user-attachments/assets/b47c624d-c4a3-473d-b2e0-b540ca17b9ba" />
Since it´s a first test i just printed a quick mounting bracket to the boom and wired it into the 12V light system of the sprayer, for more test i plan to use a Makita 18V battery to power it

Hope to see this community grow!

Kind regards,

Moritz.

**Comment by clanpaulmacgregor** [comment]
<img width="2976" height="2976" alt="IMG_20260618_1908119" src="https://github.com/user-attachments/assets/b4ed50f5-0932-41d9-a9c2-ec12670a8f3c" />
<img width="2976" height="2976" alt="IMG_20260618_1906222" src="https://github.com/user-attachments/assets/a507ee1c-b5d2-40b1-a74c-e3707b70f498" />
got the network version setup in the basement. Remote adjusting is ideal. hope to get into the real world soon

P

**Reply by geezacoleman** [reply]
That's looking great! Glad to hear you got it working.

Is there anything you would need/suggest for the controller interface? I'm just working on updating the config mechanism so it's a bit clearer -  current one is confusing. There should also be integration soon with https://noktura.tech so you can remotely manage and download data from them if they have an internet connection.

I've also added a 'Geometry' section where you can better select actuation zones (yellow) and crop images so you get the correct overlap (green).

<img width="798" height="501" alt="image" src="https://github.com/user-attachments/assets/6cc65b34-02f6-4f5f-b0fc-fab0210d3f6f" />

You can try out the new design on the [docs ](https://docs.openweedlocator.org/en/latest/controllers/wireless/networked.html) if you're interested.

It would be great to share this on the [Community Builds](https://community.openweedlocator.org/c/community-builds/9) page too.

**Comment by clanpaulmacgregor** [comment]
Hi Guy

Great work you have been doing, I only wish I was further ahead on
implementation.
The recent controller and software upgrade are a real advancement.
A couple comments
1/ I missed that a usb storage device was needed to have the images show up
in the controller preview. No big deal. I got the micro usb that still fits
into my aluminum cases.
2/ Sometimes all of the "owls" don't show up in the controller screen. So I
don't know if it is because I am using an older wifi router or ?. So if
there was a "ping" or network discovery thing to confirm connectivity? The
router has the MAC id and have stationary IPs.

The controller uses a touch screen without a physical keyboard. So some
touch screen app to confirm connection?

The actuation zone I think is key to row crop, corn and IP soys/edibles.
For example, my corn is planted on 30" rows, soys on 15", this year I got
behind and got my neighbour to plant, he uses 20" rows.
Soys are sometimes planted on 7.5" rows, in addition to winter wheat.

I read an article (somewhat negative), which said, targeted spraying will
not replace broad application of herbicide. And I believe they are correct.

The part that targeted spraying will thrive, is for escapes and weeds that
did not respond to the application of herbicide.

I recall many years ago when I was young, the role was to go out and spray
these misses. Now no one can afford the labour to do that now, so then the
answer is to apply another broad application to target these few weeds.
Expensive!

I have been searching for a semi-automated rover, something that could go
and scout out these misses/escapes, but I have yet to find anything. I was
hoping the "Acorn" rover could work, but it seems to not be active any
longer.So if you know of a source let me know. I think Australia is further
ahead on this than anywhere else in the world.

My four owls use the global shutter camera and I also have AI hats, but not
connected, so if further instruction/direction becomes available, let me
know.

Also, still searching for inexpensive solenoids, I did buy some fuel
injectors, but have not experimented with them yet. Also see weed
eradication using microwave oven parts.

Any thoughts of using a directional  X Y nozzle in place of many solenoids?
I look at these guys that use directional nozzles on model rocketry and
wonder if could work. It would be similar to the "Galvo" they use for
directional lasers. Laser light show units.

Again gret work by you and your team.

P

**Comment by clanpaulmacgregor** [comment]
#response_container_BBPPID{font-family: initial; font-size:initial; color: initial;}  Hi guyI should have read further in the docs. Problems solvedDo you know when the geometry option will be ready for the controller. I would like to have a owl view 40inches as soys are at 20inches. So need to ignore the centre section. Then 2 nozzles in 20inchesP                                                                                                                                             Sent from my BlackBerry — the most secure mobile device — via the Bell Network                                                                                                                                                                           From: ***@***.***: June 19, 2026 3:20 AMTo: ***@***.*** to: ***@***.***: ***@***.***; ***@***.***: Re: [geezacoleman/OpenWeedLocator] OWL Community Builds (Discussion #75)  
That's looking great! Glad to hear you got it working.
Is there anything you would need/suggest for the controller interface? I'm just working on updating the config mechanism so it's a bit clearer -  current one is confusing. There should also be integration soon with https://noktura.tech so you can remotely manage and download data from them if they have an internet connection.
I've also added a 'Geometry' section where you can better select actuation zones (yellow) and crop images so you get the correct overlap (green).
image.png (view on web)
